components/automate-cli/cmd/chef-automate/start_linx.go

```diff
@@ -1,67 +1,272 @@
+			return err
+		}
+		sshUser := infra.Outputs.SSHUser.Value
+		sskKeyFile := infra.Outputs.SSHKeyFile.Value
+		sshPort := infra.Outputs.SSHPort.Value
+		sshConfig := &SSHConfig{
+			sshUser:    sshUser,
+			sshKeyFile: sskKeyFile,
+			sshPort:    sshPort,
+		}
+		sshUtil := NewSSHUtil(sshConfig)
+		errorList := list.New()
+		if startCmdFlags.automate {
+			frontendIps := infra.Outputs.AutomatePrivateIps.Value
+			err = checkNodes(args, sshUtil, frontendIps, "automate", writer)
+			if err != nil {
+				errorList.PushBack(err.Error())
+			}
+		}
+		if startCmdFlags.chef_server {
+			frontendIps := infra.Outputs.ChefServerPrivateIps.Value
+			err = checkNodes(args, sshUtil, frontendIps, "chef-server", writer)
+			if err != nil {
+				errorList.PushBack(err.Error())
+			}
+		}
+		if startCmdFlags.opensearch {
+			backendIps := infra.Outputs.OpensearchPrivateIps.Value
+			err = checkNodes(args, sshUtil, backendIps, "opensearch", writer)
+			if err != nil {
+				errorList.PushBack(err.Error())
+			}
+		}
+		if startCmdFlags.postgresql {
+			backendIps := infra.Outputs.PostgresqlPrivateIps.Value
+			err = checkNodes(args, sshUtil, backendIps, "postgresql", writer)
+			return err
+		}
+		if errorList.Len() > 0 {
+			return getSingleErrorFromList(errorList)
 		}
+		return nil
+	} else {
+		writer.Title("Starting Chef Automate")
+		if isDevMode() {
+			if err := exec.Command("hab", "sup", "status").Run(); err == nil {
+				return nil
+			}
 
-		if err := os.MkdirAll("/hab/sup/default", 0755); err != nil {
-			return status.Annotate(err, status.FileAccessError)
+			if err := os.MkdirAll("/hab/sup/default", 0755); err != nil {
+				return status.Annotate(err, status.FileAccessError)
+			}
+			writer.Title("Launching the Habitat Supervisor in the background...")
+			out, err := os.Create("/hab/sup/default/sup.log")
+			if err != nil {
+				return status.Annotate(err, status.FileAccessError)
+			}
+			startSupCmd := exec.Command("hab", "sup", "run")
+			startSupCmd.Env = os.Environ()
+			startSupCmd.Env = append(startSupCmd.Env,
+				"DS_DEV=true",
+				"CHEF_AUTOMATE_SKIP_SYSTEMD=true",
+				"HAB_LICENSE=accept-no-persist",
+			)
+			startSupCmd.Stdout = out
+			startSupCmd.Stderr = out
+			startSupCmd.SysProcAttr = &syscall.SysProcAttr{
+				Setpgid: true,
+			}
+			if err := startSupCmd.Start(); err != nil {
+				return status.Annotate(err, status.HabCommandError)
+			}
+		} else {
+			systemctlCmd := exec.Command("systemctl", "start", "chef-automate.service")
+			systemctlCmd.Stdout = os.Stdout
+			systemctlCmd.Stderr = os.Stderr
+			if err := systemctlCmd.Run(); err != nil {
+				return status.Annotate(err, status.ServiceStartError)
+			}
 		}
-		writer.Title("Launching the Habitat Supervisor in the background...")
-		out, err := os.Create("/hab/sup/default/sup.log")
+	}
+	return nil
+}
+
+func startFrontEndNodes(args []string, sshUtil SSHUtil, ips []string, remoteService string, cliWriter *cli.Writer) error {
+	resultChan := make(chan Result, len(ips))
```

Comment:
Since there are no go-routines, I don't think this channel is needed. You can simplify the code by creating a list to store results.